api/python/provisioner/commands/upgrade/get_swupgrade_info.py

```diff
@@ -0,0 +1,142 @@
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+import logging
+from typing import Type
+
+from packaging import version
+from provisioner import inputs
+from provisioner.config import (CORTX_ISO_DIR, REPO_CANDIDATE_NAME,
+                                SWUpgradeInfoFields)
+
+from provisioner.salt import local_minion_id, cmd_run
+from provisioner.commands import CommandParserFillerMixin
+from provisioner.vendor import attr
+
+from . import SWUpgrade
+
+
+logger = logging.getLogger(__name__)
+
+
+@attr.s(auto_attribs=True)
+class GetSWUpgradeInfoArgs:
+    release: str = attr.ib(
+        metadata={
+            inputs.METADATA_ARGPARSER: {
+                'help': "SW upgrade repository release version"
+            }
+        },
+        default=None
+    )
+
+
+class GetSWUpgradeInfo(CommandParserFillerMixin):
+    """
+    Base class that provides API for getting SW upgrade repository information.
+
+    """
+
+    input_type: Type[inputs.NoParams] = inputs.NoParams
+    _run_args_type = GetSWUpgradeInfoArgs
+
+    @staticmethod
+    def _get_package_version(release: str) -> dict:
+        """
+        Function returns information about CORTX packages and their versions.
+
+        Parameters
+        ----------
+        release : str
+            SW upgrade repository version
+
+        Returns
+        -------
+        dict
+            return dictionary with CORTX packages and their versions
+
+        """
+        repo = f"sw_upgrade_{CORTX_ISO_DIR}_{release}"
+
+        cmd = (f"yum repo-pkgs {repo} list 2>/dev/null | "
+               f"grep '{repo}' | awk '{{print $1\" \"$2}}'")
+
+        res = cmd_run(cmd, targets=local_minion_id(),
+                      fun_kwargs=dict(python_shell=True))
+
+        packages = res[local_minion_id()].strip().split('\n')
+
+        if packages:
+            logger.debug(f"List of packages in repository '{repo}':"
+                         f" {packages}")
+        else:
+            logger.debug(f"There are no packages in repository '{repo}'")
+
+        res = dict()
+        # NOTE: Format is following
+        # ```
+        #  {
+        #      'cortx-motr': {
+        #             'version': '2.0.0-277',
+        #          },
+        #  }
+        # ```
+        #
+        # TODO: Along the with 'version', field we need to add
```

Comment:
please add a reference to Jira task here

Reply:
Done